pkgs/applications/version-management/mercurial/4.9.nix

```diff
@@ -10,7 +10,7 @@
 }:
 
 let
-  inherit (python2Packages) docutils hg-git dulwich python;
+  inherit (python2Packages) docutils hg-git-0.8.2 dulwich python;
```

Comment:
Please put the version in an inherit in top level and leave hg-git here.

Reply:
sorry, not sure I understand I am fairly new to nix packaging. Like so?

```
diff --git a/pkgs/applications/version-management/mercurial/4.9.nix b/pkgs/applications/version-management/mercurial/4.9.nix
index 472ea0b7c20..3379f6a4138 100644
--- a/pkgs/applications/version-management/mercurial/4.9.nix
+++ b/pkgs/applications/version-management/mercurial/4.9.nix
@@ -1,6 +1,7 @@
 { lib, stdenv, fetchurl, python2Packages, makeWrapper, unzip
 , guiSupport ? false, tk ? null
 , ApplicationServices
+, hg-git ? hg-git-0.8
 , mercurialSrc ? fetchurl rec {
     meta.name = "mercurial-${meta.version}";
     meta.version = "4.9.1";
@@ -10,7 +11,7 @@
 }:

 let
-  inherit (python2Packages) docutils hg-git-0.8.2 dulwich python;
+  inherit (python2Packages) docutils hg-git dulwich python;

 in python2Packages.buildPythonApplication {

@@ -24,7 +25,7 @@ in python2Packages.buildPythonApplication {
   buildInputs = [ makeWrapper docutils unzip ]
     ++ lib.optionals stdenv.isDarwin [ ApplicationServices ];

-  propagatedBuildInputs = [ hg-git-0.8.12 dulwich ];
+  propagatedBuildInputs = [ hg-git dulwich ];

   makeFlags = [ "PREFIX=$(out)" ];
   ```

Reply:
In all-packages like

```
mercurial = callPackage ../applications/version-management/mercurial/4.9.nix {
  hg-git = hg-git-0.8;                                                                                                                                  };
 ```

then you don't need to do anything in ``pkgs/applications/version-management/mercurial/4.9.nix``.